modules/account-lib/src/coin/stx/transactionBuilder.ts

```diff
@@ -0,0 +1,267 @@
+import BigNumber from 'bignumber.js';
+import BigNum from 'bn.js';
+import { BaseCoin as CoinConfig } from '@bitgo/statics/dist/src/base';
+import {
+  BufferReader,
+  deserializeTransaction,
+  emptyMessageSignature,
+  isSingleSig,
+  makeSigHashPreSign,
+  publicKeyFromSignature,
+} from '@stacks/transactions';
+import { StacksNetwork, StacksTestnet } from '@stacks/network';
+import { BaseTransactionBuilder } from '../baseCoin';
+import {
+  BuildTransactionError,
+  ParseTransactionError,
+  SigningError,
+  InvalidTransactionError,
+} from '../baseCoin/errors';
+import { BaseAddress, BaseFee, BaseKey } from '../baseCoin/iface';
+import { Transaction } from './transaction';
+import { KeyPair } from './keyPair';
+import { SignatureData } from './iface';
+import { isValidAddress, removeHexPrefix } from './utils';
+
+export abstract class TransactionBuilder extends BaseTransactionBuilder {
+  private _transaction: Transaction;
+  protected _fee: BaseFee;
+  protected _nonce: number;
+  protected _source: BaseAddress;
+  protected _memo: string;
+  protected _numberSignatures: number;
+  protected _multiSignerKeyPairs: KeyPair[];
+  protected _signatures: SignatureData[];
+  protected _network: StacksNetwork;
+
+  constructor(_coinConfig: Readonly<CoinConfig>) {
+    super(_coinConfig);
+    this._multiSignerKeyPairs = [];
+    this._signatures = [];
+    this._numberSignatures = 1;
```

Comment:
Why is this initialized to 1? Seems like 0 would be a better default candidate

Reply:
would have to adjust `buildImplementation` too

Reply:
I used 1, as minimum 1 sign should be required for signed tx. Updated to 0. so now `_numberSignatures` gets updated value in `signImplementation` method